tsec-libsodium/src/test/scala/tsec/libsodium/authentication/OnetimeAuthTest.scala

```diff
@@ -0,0 +1,38 @@
+package tsec.libsodium.authentication
+
+import cats.effect.IO
+import tsec.common._
+import tsec.libsodium.SodiumSpec
+
+class OnetimeAuthTest extends SodiumSpec {
+
+  behavior of "One-Time authentication"
+
+  it should "verify message properly" in {
+    forAll { (s: String) =>
+      val program: IO[Boolean] = for {
+        key <- OnetimeAuth.generateKey[IO]
+        message = s.utf8Bytes
+        tag <- OnetimeAuth.generateTag[IO](key, message)
+        isOk   <- OnetimeAuth.verify[IO](key, message, tag)
+      } yield isOk
+
+      program.unsafeRunSync() mustBe trueg
+    }
+  }
+
+  it should "not verify manipulated message" in {
+    forAll { (s: String) =>
+      val program: IO[Boolean] = for {
+        key <- OnetimeAuth.generateKey[IO]
+        message  = s.utf8Bytes
+        message2 = ("$" + s).utf8Bytes
```

Comment:
I would do a `forAll { (s1, s2) => s1 == s2 || program.unsafeRunSync() mustBe false }`.

Reply:
I used `whenever` now, thanks for the help :+1: